{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "productive-ordinary",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#the following codes create files with the SNPs list for each chromosome\n",
    "cat > /teams/CSE284_SP21_A00/team9/snps16.list\n",
    "# MC1R_R SNPs \n",
    "rs201326893 #Y152OCH   \n",
    "rs796296176 #N29insA \n",
    "rs1805006\n",
    "rs11547464\n",
    "rs1805007\n",
    "rs1805008\n",
    "rs1805009\n",
    "#MC1R_r SNPs \n",
    "rs1805005\n",
    "rs2228479\n",
    "rs1110400\n",
    "rs885479"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "occasional-spyware",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat > /teams/CSE284_SP21_A00/team9/snps15.list\n",
    "rs12913832 \n",
    "rs1800407 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "previous-album",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat > /teams/CSE284_SP21_A00/team9/snps6.list\n",
    "rs4959270\n",
    "rs12203592 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "referenced-matrix",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat > /teams/CSE284_SP21_A00/team9/snps11.list\n",
    "rs1042602  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "parliamentary-indonesia",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat > /teams/CSE284_SP21_A00/team9/snps5.list\n",
    "rs28777  \n",
    "rs16891982"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "hindu-talent",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat > /teams/CSE284_SP21_A00/team9/snps9.list\n",
    "rs683  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "authentic-shooting",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat > /teams/CSE284_SP21_A00/team9/snps14.list\n",
    "rs2402130  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "amino-strength",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat > /teams/CSE284_SP21_A00/team9/snps12.list\n",
    "rs12821256"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "floral-element",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat > /teams/CSE284_SP21_A00/team9/snps20.list\n",
    "rs2378249"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "dense-answer",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "bash: line 1: DIREC1000: command not found\n",
      "bash: line 2: DIRECOUT: command not found\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /datasets/cs284-sp21-A00-public/1000Genomes/ALL.chr16.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\t--out /teams/CSE284_SP21_A00/team9/Filtered_hair_SNPs_16\n",
      "\t--recode\n",
      "\t--snps /teams/CSE284_SP21_A00/team9/snps16.list\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 10 out of a possible 2697949 Sites\n",
      "Run Time = 119.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /datasets/cs284-sp21-A00-public/1000Genomes/ALL.chr15.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\t--out /teams/CSE284_SP21_A00/team9/Filtered_hair_SNPs_15\n",
      "\t--recode\n",
      "\t--snps /teams/CSE284_SP21_A00/team9/snps15.list\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 2 out of a possible 2424689 Sites\n",
      "Run Time = 108.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /datasets/cs284-sp21-A00-public/1000Genomes/ALL.chr6.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\t--out /teams/CSE284_SP21_A00/team9/Filtered_hair_SNPs_6\n",
      "\t--recode\n",
      "\t--snps /teams/CSE284_SP21_A00/team9/snps6.list\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 2 out of a possible 5024119 Sites\n",
      "Run Time = 212.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /datasets/cs284-sp21-A00-public/1000Genomes/ALL.chr11.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\t--out /teams/CSE284_SP21_A00/team9/Filtered_hair_SNPs_11\n",
      "\t--recode\n",
      "\t--snps /teams/CSE284_SP21_A00/team9/snps11.list\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 1 out of a possible 4045628 Sites\n",
      "Run Time = 167.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /datasets/cs284-sp21-A00-public/1000Genomes/ALL.chr6.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\t--out /teams/CSE284_SP21_A00/team9/Filtered_hair_SNPs_6\n",
      "\t--recode\n",
      "\t--snps /teams/CSE284_SP21_A00/team9/snps6.list\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 2 out of a possible 5024119 Sites\n",
      "Run Time = 220.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /datasets/cs284-sp21-A00-public/1000Genomes/ALL.chr5.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\t--out /teams/CSE284_SP21_A00/team9/Filtered_hair_SNPs_5\n",
      "\t--recode\n",
      "\t--snps /teams/CSE284_SP21_A00/team9/snps5.list\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 2 out of a possible 5265763 Sites\n",
      "Run Time = 216.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /datasets/cs284-sp21-A00-public/1000Genomes/ALL.chr9.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\t--out /teams/CSE284_SP21_A00/team9/Filtered_hair_SNPs_9\n",
      "\t--recode\n",
      "\t--snps /teams/CSE284_SP21_A00/team9/snps9.list\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 1 out of a possible 3560687 Sites\n",
      "Run Time = 155.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /datasets/cs284-sp21-A00-public/1000Genomes/ALL.chr14.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\t--out /teams/CSE284_SP21_A00/team9/Filtered_hair_SNPs_14\n",
      "\t--recode\n",
      "\t--snps /teams/CSE284_SP21_A00/team9/snps14.list\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 1 out of a possible 2655067 Sites\n",
      "Run Time = 116.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /datasets/cs284-sp21-A00-public/1000Genomes/ALL.chr12.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\t--out /teams/CSE284_SP21_A00/team9/Filtered_hair_SNPs_12\n",
      "\t--recode\n",
      "\t--snps /teams/CSE284_SP21_A00/team9/snps12.list\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 1 out of a possible 3868428 Sites\n",
      "Run Time = 169.00 seconds\n",
      "\n",
      "VCFtools - UNKNOWN\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf /datasets/cs284-sp21-A00-public/1000Genomes/ALL.chr20.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\t--out /teams/CSE284_SP21_A00/team9/Filtered_hair_SNPs_20\n",
      "\t--recode\n",
      "\t--snps /teams/CSE284_SP21_A00/team9/snps20.list\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "After filtering, kept 1 out of a possible 1812841 Sites\n",
      "Run Time = 76.00 seconds\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "DIREC1000 =  /datasets/cs284-sp21-A00-public/1000Genomes\n",
    "DIRECOUT = /teams/CSE284_SP21_A00/team9/\n",
    "\n",
    "for chrom in 16 15 6 11 6 5 9 14 12 20\n",
    "do\n",
    "    vcftools --gzvcf /datasets/cs284-sp21-A00-public/1000Genomes/ALL.chr${chrom}.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --out /teams/CSE284_SP21_A00/team9/Filtered_hair_SNPs_${chrom} --snps /teams/CSE284_SP21_A00/team9/snps${chrom}.list --recode\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "stock-workstation",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>rsid</th>\n",
       "      <th>Minor allele</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>16</td>\n",
       "      <td>MC1R_R</td>\n",
       "      <td>R</td>\n",
       "      <td>1.11</td>\n",
       "      <td>0.55</td>\n",
       "      <td>4.09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>16</td>\n",
       "      <td>MC1R_r</td>\n",
       "      <td>r</td>\n",
       "      <td>0.46</td>\n",
       "      <td>0.55</td>\n",
       "      <td>0.61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15</td>\n",
       "      <td>rs12913832</td>\n",
       "      <td>T</td>\n",
       "      <td>-1.75</td>\n",
       "      <td>0.10</td>\n",
       "      <td>-2.49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>rs12203592</td>\n",
       "      <td>T</td>\n",
       "      <td>-1.29</td>\n",
       "      <td>-1.15</td>\n",
       "      <td>-1.13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>11</td>\n",
       "      <td>rs1042602</td>\n",
       "      <td>A</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.30</td>\n",
       "      <td>1.20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>rs4959270</td>\n",
       "      <td>A</td>\n",
       "      <td>0.77</td>\n",
       "      <td>0.85</td>\n",
       "      <td>1.15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>5</td>\n",
       "      <td>rs28777</td>\n",
       "      <td>C</td>\n",
       "      <td>-1.69</td>\n",
       "      <td>-13.89</td>\n",
       "      <td>0.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>9</td>\n",
       "      <td>rs683</td>\n",
       "      <td>C</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.58</td>\n",
       "      <td>-0.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>15</td>\n",
       "      <td>rs1800407</td>\n",
       "      <td>T</td>\n",
       "      <td>0.49</td>\n",
       "      <td>-1.14</td>\n",
       "      <td>0.19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>14</td>\n",
       "      <td>rs2402130</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.48</td>\n",
       "      <td>-0.09</td>\n",
       "      <td>-0.54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>12</td>\n",
       "      <td>rs12821256</td>\n",
       "      <td>C</td>\n",
       "      <td>0.69</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>5</td>\n",
       "      <td>rs16891982</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.82</td>\n",
       "      <td>-11.78</td>\n",
       "      <td>-3.48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>20</td>\n",
       "      <td>rs2378249</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.18</td>\n",
       "      <td>-0.16</td>\n",
       "      <td>0.40</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Chromosome        rsid Minor allele    1     2    3\n",
       "0           16      MC1R_R            R  1.11   0.55  4.09\n",
       "1           16      MC1R_r            r  0.46   0.55  0.61\n",
       "2           15  rs12913832            T -1.75   0.10 -2.49\n",
       "3            6  rs12203592            T -1.29  -1.15 -1.13\n",
       "4           11   rs1042602            A  0.39   0.30  1.20\n",
       "5            6   rs4959270            A  0.77   0.85  1.15\n",
       "6            5     rs28777            C -1.69 -13.89  0.10\n",
       "7            9       rs683            C  0.10   0.58 -0.02\n",
       "8           15   rs1800407            T  0.49  -1.14  0.19\n",
       "9           14   rs2402130            G -0.48  -0.09 -0.54\n",
       "10          12  rs12821256            C  0.69   0.01  0.87\n",
       "11           5  rs16891982            C -0.82 -11.78 -3.48\n",
       "12          20   rs2378249            G -0.18  -0.16  0.40"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#need to create bed file with info from the paper \n",
    "import pandas as pd\n",
    "\n",
    "rsid = ['MC1R_R', 'MC1R_r','rs12913832','rs12203592','rs1042602', 'rs4959270', 'rs28777', 'rs683','rs1800407','rs2402130','rs12821256','rs16891982','rs2378249']\n",
    "bed_file_concentrated = pd.DataFrame(columns = ['Chromosome', 'rsid', 'Minor allele', '1', '2', '3'])\n",
    "chrom = [16,16,15,6,11,6,5,9,15,14,12,5,20]\n",
    "bed_file_concentrated['Chromosome'] = chrom\n",
    "bed_file_concentrated['rsid'] = rsid\n",
    "bed_file_concentrated['Minor allele'] = ['R','r','T','T','A','A','C','C','T','G','C','C','G']\n",
    "bed_file_concentrated['1'] = [1.11,0.46,-1.75,-1.29,0.39,0.77,-1.69,0.1,0.49,-0.48,0.69,-0.82,-0.18]\n",
    "bed_file_concentrated['2'] = [0.55,0.55,0.1,-1.15, 0.30,0.85,-13.89,0.58,-1.14,-0.09,0.01,-11.78,-0.16]\n",
    "bed_file_concentrated['3'] = [4.09,0.61,-2.49,-1.13,1.2,1.15,0.1,-0.02,0.19,-0.54,0.87,-3.48,0.4]\n",
    "\n",
    "bed_file_concentrated\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "commercial-broad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>rsid</th>\n",
       "      <th>Minor allele</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>16</td>\n",
       "      <td>rs201326893</td>\n",
       "      <td>A</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>16</td>\n",
       "      <td>rs796296176</td>\n",
       "      <td>A</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>16</td>\n",
       "      <td>rs1805006</td>\n",
       "      <td>A</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>16</td>\n",
       "      <td>rs11547464</td>\n",
       "      <td>A</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>16</td>\n",
       "      <td>rs1805007</td>\n",
       "      <td>T</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>16</td>\n",
       "      <td>rs1805008</td>\n",
       "      <td>T</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>16</td>\n",
       "      <td>rs1805009</td>\n",
       "      <td>C</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>16</td>\n",
       "      <td>rs1805005</td>\n",
       "      <td>T</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>16</td>\n",
       "      <td>rs2228479</td>\n",
       "      <td>A</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>16</td>\n",
       "      <td>rs1110400</td>\n",
       "      <td>C</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>16</td>\n",
       "      <td>rs885479</td>\n",
       "      <td>A</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>15</td>\n",
       "      <td>rs12913832</td>\n",
       "      <td>T</td>\n",
       "      <td>-1.75</td>\n",
       "      <td>0.10</td>\n",
       "      <td>-2.49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>6</td>\n",
       "      <td>rs12203592</td>\n",
       "      <td>T</td>\n",
       "      <td>-1.29</td>\n",
       "      <td>-1.15</td>\n",
       "      <td>-1.13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>11</td>\n",
       "      <td>rs1042602</td>\n",
       "      <td>A</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.30</td>\n",
       "      <td>1.20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>6</td>\n",
       "      <td>rs4959270</td>\n",
       "      <td>A</td>\n",
       "      <td>0.77</td>\n",
       "      <td>0.85</td>\n",
       "      <td>1.15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>5</td>\n",
       "      <td>rs28777</td>\n",
       "      <td>C</td>\n",
       "      <td>-1.69</td>\n",
       "      <td>-13.89</td>\n",
       "      <td>0.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>9</td>\n",
       "      <td>rs683</td>\n",
       "      <td>C</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.58</td>\n",
       "      <td>-0.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>15</td>\n",
       "      <td>rs1800407</td>\n",
       "      <td>T</td>\n",
       "      <td>0.49</td>\n",
       "      <td>-1.14</td>\n",
       "      <td>0.19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>14</td>\n",
       "      <td>rs2402130</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.48</td>\n",
       "      <td>-0.09</td>\n",
       "      <td>-0.54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>12</td>\n",
       "      <td>rs12821256</td>\n",
       "      <td>C</td>\n",
       "      <td>0.69</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>5</td>\n",
       "      <td>rs16891982</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.82</td>\n",
       "      <td>-11.78</td>\n",
       "      <td>-3.48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>20</td>\n",
       "      <td>rs2378249</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.18</td>\n",
       "      <td>-0.16</td>\n",
       "      <td>0.40</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Chromosome         rsid Minor allele    1     2    3\n",
       "0           16  rs201326893            A  0.00   0.00  0.00\n",
       "1           16  rs796296176            A  0.00   0.00  0.00\n",
       "2           16    rs1805006            A  0.00   0.00  0.00\n",
       "3           16   rs11547464            A  0.00   0.00  0.00\n",
       "4           16    rs1805007            T  0.00   0.00  0.00\n",
       "5           16    rs1805008            T  0.00   0.00  0.00\n",
       "6           16    rs1805009            C  0.00   0.00  0.00\n",
       "7           16    rs1805005            T  0.00   0.00  0.00\n",
       "8           16    rs2228479            A  0.00   0.00  0.00\n",
       "9           16    rs1110400            C  0.00   0.00  0.00\n",
       "10          16     rs885479            A  0.00   0.00  0.00\n",
       "11          15   rs12913832            T -1.75   0.10 -2.49\n",
       "12           6   rs12203592            T -1.29  -1.15 -1.13\n",
       "13          11    rs1042602            A  0.39   0.30  1.20\n",
       "14           6    rs4959270            A  0.77   0.85  1.15\n",
       "15           5      rs28777            C -1.69 -13.89  0.10\n",
       "16           9        rs683            C  0.10   0.58 -0.02\n",
       "17          15    rs1800407            T  0.49  -1.14  0.19\n",
       "18          14    rs2402130            G -0.48  -0.09 -0.54\n",
       "19          12   rs12821256            C  0.69   0.01  0.87\n",
       "20           5   rs16891982            C -0.82 -11.78 -3.48\n",
       "21          20    rs2378249            G -0.18  -0.16  0.40"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#need to create bed file with info from the paper \n",
    "import pandas as pd\n",
    "\n",
    "rsid = ['rs201326893', 'rs796296176', 'rs1805006', 'rs11547464' , 'rs1805007', 'rs1805008', 'rs1805009', 'rs1805005','rs2228479', 'rs1110400', 'rs885479', 'rs12913832','rs12203592','rs1042602', 'rs4959270', 'rs28777', 'rs683','rs1800407','rs2402130','rs12821256','rs16891982','rs2378249']\n",
    "bed_file = pd.DataFrame(columns = ['Chromosome', 'rsid', 'Minor allele', '1', '2', '3'])\n",
    "chrom = [16,16,16,16,16,16,16,16,16,16,16,15,6,11,6,5,9,15,14,12,5,20]\n",
    "bed_file['Chromosome'] = chrom\n",
    "bed_file['rsid'] = rsid\n",
    "bed_file['Minor allele'] = ['A','A','A','A','T','T','C','T','A','C','A','T','T','A','A','C','C','T','G','C','C','G']\n",
    "bed_file['1'] = [0,0,0,0,0,0,0,0,0,0,0,-1.75,-1.29,0.39,0.77,-1.69,0.1,0.49,-0.48,0.69,-0.82,-0.18]\n",
    "bed_file['2'] = [0,0,0,0,0,0,0,0,0,0,0,0.10,-1.15, 0.30,0.85,-13.89,0.58,-1.14,-0.09,0.01,-11.78,-0.16]\n",
    "bed_file['3'] = [0,0,0,0,0,0,0,0,0,0,0,-2.49,-1.13,1.2,1.15,0.1,-0.02,0.19,-0.54,0.87,-3.48,0.4]\n",
    "\n",
    "bed_file "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "auburn-affiliate",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "#extract only key info and put into output file \n",
    "bcftools query -f '[ %CHROM %ID %REF %ALT %GT \\n]' /teams/CSE284_SP21_A00/team9/Filtered_hair_SNPs_All.vcf > /teams/CSE284_SP21_A00/team9/all_genotypes_hair_SNP.txt\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "finite-diana",
   "metadata": {},
   "outputs": [],
   "source": [
    "#make dataframe with all the info\n",
    "all_GT = pd.read_csv('/teams/CSE284_SP21_A00/team9/all_genotypes_hair_SNP.txt', delimiter = \" \", header = None)\n",
    "all_GT = all_GT.drop([0,6],axis = 1)\n",
    "all_GT.columns = ['chrom','rsid','REF','ALT','GT']\n",
    "#attach sample names to the dataframe\n",
    "all_GT.insert(0, 'sample', 0) #insert sample column at the beginning"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "first-philadelphia",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>chrom</th>\n",
       "      <th>rsid</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>GT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>rs1042602</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>rs1042602</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>rs1042602</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>rs1042602</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>rs1042602</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50075</th>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>rs1805009</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50076</th>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>rs1805009</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50077</th>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>rs1805009</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50078</th>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>rs1805009</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50079</th>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>rs1805009</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0|0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>50080 rows  6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       sample  chrom       rsid REF ALT   GT\n",
       "0           0     11  rs1042602   C   A  0|0\n",
       "1           0     11  rs1042602   C   A  0|0\n",
       "2           0     11  rs1042602   C   A  0|0\n",
       "3           0     11  rs1042602   C   A  0|0\n",
       "4           0     11  rs1042602   C   A  0|0\n",
       "...       ...    ...        ...  ..  ..  ...\n",
       "50075       0     16  rs1805009   G   C  0|0\n",
       "50076       0     16  rs1805009   G   C  0|0\n",
       "50077       0     16  rs1805009   G   C  0|0\n",
       "50078       0     16  rs1805009   G   C  0|0\n",
       "50079       0     16  rs1805009   G   C  0|0\n",
       "\n",
       "[50080 rows x 6 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_GT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "overhead-reply",
   "metadata": {},
   "outputs": [],
   "source": [
    "#get samples names \n",
    "import pandas as pd\n",
    "samples_list = pd.read_csv('samples_ID.txt', sep='\\t', header=None, error_bad_lines=False)\n",
    "samples_list = samples_list.T[0]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "impaired-vegetable",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.8/site-packages/pandas/core/indexing.py:1636: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self._setitem_single_block(indexer, value, name)\n"
     ]
    }
   ],
   "source": [
    "SNPs_total = len(set(all_GT['rsid']))\n",
    "number_samples = len(samples_list)\n",
    "\n",
    "start_idx = 0\n",
    "for i in range(0,SNPs_total):\n",
    "    c = 0 \n",
    "    for b in range(0+start_idx,len(samples_list)+start_idx):\n",
    "        all_GT['sample'].iloc[b] = samples_list[c]\n",
    "        c = c+1 \n",
    "    start_idx = start_idx + number_samples\n",
    "\n",
    "    \n",
    "#translate genotypes into 0,1,2 information\n",
    "value = []\n",
    "for i in range(len(all_GT)):  \n",
    "    rsid = all_GT['rsid'].iloc[i]  #get rsid \n",
    "    minor_allele = bed_file[bed_file['rsid'] == rsid].iloc[0,2] #find corresponding minor allele\n",
    "    if all_GT['GT'].iloc[i] == '1|1':\n",
    "        if minor_allele == all_GT['ALT'].iloc[i]:\n",
    "            value.append(2)\n",
    "        else:\n",
    "            value.append(0)\n",
    "    elif all_GT['GT'].iloc[i] == '0|1':\n",
    "        if minor_allele == all_GT['ALT'].iloc[i] or minor_allele == all_GT['REF'].iloc[i]:\n",
    "            value.append(1)\n",
    "        else:\n",
    "            value.append(0)\n",
    "    elif all_GT['GT'].iloc[i] == '1|0':\n",
    "        if minor_allele == all_GT['ALT'].iloc[i] or minor_allele == all_GT['REF'].iloc[i]:\n",
    "            value.append(1)\n",
    "        else:\n",
    "            value.append(0)\n",
    "    elif all_GT['GT'].iloc[i] == '0|0':\n",
    "        if minor_allele == all_GT['REF'].iloc[i]:\n",
    "            value.append(2)\n",
    "        else:\n",
    "            value.append(0)\n",
    "        \n",
    "all_GT['X'] = value\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "isolated-courage",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table =  all_GT[all_GT['sample'] == 'HG00096']\n",
    "table[table['rsid'] == 'rs1042602'].X[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "decimal-storage",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>population</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HG00096</th>\n",
       "      <td>GBR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HG00097</th>\n",
       "      <td>GBR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HG00099</th>\n",
       "      <td>GBR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HG00100</th>\n",
       "      <td>GBR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HG00101</th>\n",
       "      <td>GBR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NA21137</th>\n",
       "      <td>GIH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NA21141</th>\n",
       "      <td>GIH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NA21142</th>\n",
       "      <td>GIH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NA21143</th>\n",
       "      <td>GIH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NA21144</th>\n",
       "      <td>GIH</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2504 rows  1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        population\n",
       "0                 \n",
       "HG00096        GBR\n",
       "HG00097        GBR\n",
       "HG00099        GBR\n",
       "HG00100        GBR\n",
       "HG00101        GBR\n",
       "...            ...\n",
       "NA21137        GIH\n",
       "NA21141        GIH\n",
       "NA21142        GIH\n",
       "NA21143        GIH\n",
       "NA21144        GIH\n",
       "\n",
       "[2504 rows x 1 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get population labels for all samples \n",
    "sample_to_pop = {} # maps sample -> population code. e.g. sample_to_pop[\"NA12878\"] = \"CEU\"\n",
    "with open(\"/datasets/cs284-sp21-A00-public/ps2/ps2_reference_labels.csv\", \"r\") as f:\n",
    "    for line in f:\n",
    "        sample, pop = line.strip().split(\",\")\n",
    "        sample_to_pop[sample] = pop\n",
    "\n",
    "sample_pop_match = pd.DataFrame(columns = ['population'], index = samples_list)\n",
    "\n",
    "for idx, s in enumerate(samples_list):\n",
    "    if s in sample_to_pop:  #first check if we have the label for it \n",
    "        sample_pop_match['population'].iloc[idx] = sample_to_pop[s]\n",
    "sample_pop_match      "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "processed-vertex",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>chrom</th>\n",
       "      <th>rsid</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>GT</th>\n",
       "      <th>X</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00096</td>\n",
       "      <td>11</td>\n",
       "      <td>rs1042602</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00097</td>\n",
       "      <td>11</td>\n",
       "      <td>rs1042602</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG00099</td>\n",
       "      <td>11</td>\n",
       "      <td>rs1042602</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00100</td>\n",
       "      <td>11</td>\n",
       "      <td>rs1042602</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG00101</td>\n",
       "      <td>11</td>\n",
       "      <td>rs1042602</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50075</th>\n",
       "      <td>NA21137</td>\n",
       "      <td>16</td>\n",
       "      <td>rs1805009</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50076</th>\n",
       "      <td>NA21141</td>\n",
       "      <td>16</td>\n",
       "      <td>rs1805009</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50077</th>\n",
       "      <td>NA21142</td>\n",
       "      <td>16</td>\n",
       "      <td>rs1805009</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50078</th>\n",
       "      <td>NA21143</td>\n",
       "      <td>16</td>\n",
       "      <td>rs1805009</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50079</th>\n",
       "      <td>NA21144</td>\n",
       "      <td>16</td>\n",
       "      <td>rs1805009</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>50080 rows  7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample  chrom       rsid REF ALT   GT  X\n",
       "0      HG00096     11  rs1042602   C   A  0|0  0\n",
       "1      HG00097     11  rs1042602   C   A  0|0  0\n",
       "2      HG00099     11  rs1042602   C   A  0|0  0\n",
       "3      HG00100     11  rs1042602   C   A  0|0  0\n",
       "4      HG00101     11  rs1042602   C   A  0|0  0\n",
       "...        ...    ...        ...  ..  ..  ... ..\n",
       "50075  NA21137     16  rs1805009   G   C  0|0  0\n",
       "50076  NA21141     16  rs1805009   G   C  0|0  0\n",
       "50077  NA21142     16  rs1805009   G   C  0|0  0\n",
       "50078  NA21143     16  rs1805009   G   C  0|0  0\n",
       "50079  NA21144     16  rs1805009   G   C  0|0  0\n",
       "\n",
       "[50080 rows x 7 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_GT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "dirty-anxiety",
   "metadata": {},
   "outputs": [],
   "source": [
    "# generate for testing\n",
    "sample_pop_match.to_csv(\"sample_pop_match.txt\", sep=\" \")\n",
    "all_GT.to_csv(\"1kg_hair_gt.txt\", sep=\" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "interior-success",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0         MC1R_R\n",
       "1         MC1R_r\n",
       "2     rs12913832\n",
       "3     rs12203592\n",
       "4      rs1042602\n",
       "5      rs4959270\n",
       "6        rs28777\n",
       "7          rs683\n",
       "8      rs1800407\n",
       "9      rs2402130\n",
       "10    rs12821256\n",
       "11    rs16891982\n",
       "12     rs2378249\n",
       "Name: rsid, dtype: object"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bed_file_concentrated['rsid']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "minus-picnic",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pBlond</th>\n",
       "      <th>pBrown</th>\n",
       "      <th>pRed</th>\n",
       "      <th>pBlack</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>population</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ACB</th>\n",
       "      <td>0.144143</td>\n",
       "      <td>9.191319e-04</td>\n",
       "      <td>0.000460</td>\n",
       "      <td>0.854477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASW</th>\n",
       "      <td>0.214322</td>\n",
       "      <td>6.331848e-03</td>\n",
       "      <td>0.001346</td>\n",
       "      <td>0.778001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BEB</th>\n",
       "      <td>0.147893</td>\n",
       "      <td>4.292293e-09</td>\n",
       "      <td>0.000493</td>\n",
       "      <td>0.851614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CDX</th>\n",
       "      <td>0.112380</td>\n",
       "      <td>8.855685e-13</td>\n",
       "      <td>0.000205</td>\n",
       "      <td>0.887415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CEU</th>\n",
       "      <td>0.788829</td>\n",
       "      <td>5.684941e-02</td>\n",
       "      <td>0.044110</td>\n",
       "      <td>0.110211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHB</th>\n",
       "      <td>0.104775</td>\n",
       "      <td>1.234433e-13</td>\n",
       "      <td>0.000228</td>\n",
       "      <td>0.894997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHS</th>\n",
       "      <td>0.117412</td>\n",
       "      <td>6.570803e-14</td>\n",
       "      <td>0.000253</td>\n",
       "      <td>0.882334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CLM</th>\n",
       "      <td>0.590383</td>\n",
       "      <td>3.449625e-02</td>\n",
       "      <td>0.019500</td>\n",
       "      <td>0.355621</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ESN</th>\n",
       "      <td>0.071548</td>\n",
       "      <td>6.977994e-13</td>\n",
       "      <td>0.000085</td>\n",
       "      <td>0.928367</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FIN</th>\n",
       "      <td>0.812812</td>\n",
       "      <td>5.291742e-02</td>\n",
       "      <td>0.024798</td>\n",
       "      <td>0.109473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GBR</th>\n",
       "      <td>0.789207</td>\n",
       "      <td>5.133869e-02</td>\n",
       "      <td>0.037341</td>\n",
       "      <td>0.122113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GIH</th>\n",
       "      <td>0.198740</td>\n",
       "      <td>2.789187e-12</td>\n",
       "      <td>0.001089</td>\n",
       "      <td>0.800171</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GWD</th>\n",
       "      <td>0.077460</td>\n",
       "      <td>1.005816e-12</td>\n",
       "      <td>0.000120</td>\n",
       "      <td>0.922420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IBS</th>\n",
       "      <td>0.723568</td>\n",
       "      <td>4.383728e-02</td>\n",
       "      <td>0.027501</td>\n",
       "      <td>0.205094</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ITU</th>\n",
       "      <td>0.165784</td>\n",
       "      <td>6.511905e-04</td>\n",
       "      <td>0.000727</td>\n",
       "      <td>0.832838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>JPT</th>\n",
       "      <td>0.151616</td>\n",
       "      <td>9.520952e-13</td>\n",
       "      <td>0.000230</td>\n",
       "      <td>0.848154</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KHV</th>\n",
       "      <td>0.120253</td>\n",
       "      <td>5.276638e-13</td>\n",
       "      <td>0.000334</td>\n",
       "      <td>0.879412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LWK</th>\n",
       "      <td>0.080059</td>\n",
       "      <td>6.184784e-13</td>\n",
       "      <td>0.000110</td>\n",
       "      <td>0.919831</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSL</th>\n",
       "      <td>0.101333</td>\n",
       "      <td>5.837016e-08</td>\n",
       "      <td>0.000120</td>\n",
       "      <td>0.898547</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MXL</th>\n",
       "      <td>0.402906</td>\n",
       "      <td>1.738450e-02</td>\n",
       "      <td>0.008188</td>\n",
       "      <td>0.571521</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PEL</th>\n",
       "      <td>0.230073</td>\n",
       "      <td>4.555846e-03</td>\n",
       "      <td>0.001854</td>\n",
       "      <td>0.763516</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PJL</th>\n",
       "      <td>0.160604</td>\n",
       "      <td>5.917921e-08</td>\n",
       "      <td>0.000739</td>\n",
       "      <td>0.838656</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PUR</th>\n",
       "      <td>0.490955</td>\n",
       "      <td>3.516436e-02</td>\n",
       "      <td>0.015530</td>\n",
       "      <td>0.458350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STU</th>\n",
       "      <td>0.160275</td>\n",
       "      <td>1.486042e-03</td>\n",
       "      <td>0.001155</td>\n",
       "      <td>0.837084</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TSI</th>\n",
       "      <td>0.796755</td>\n",
       "      <td>6.499689e-02</td>\n",
       "      <td>0.039268</td>\n",
       "      <td>0.098980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>YRI</th>\n",
       "      <td>0.076407</td>\n",
       "      <td>9.138039e-13</td>\n",
       "      <td>0.000099</td>\n",
       "      <td>0.923494</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              pBlond        pBrown      pRed    pBlack\n",
       "population                                            \n",
       "ACB         0.144143  9.191319e-04  0.000460  0.854477\n",
       "ASW         0.214322  6.331848e-03  0.001346  0.778001\n",
       "BEB         0.147893  4.292293e-09  0.000493  0.851614\n",
       "CDX         0.112380  8.855685e-13  0.000205  0.887415\n",
       "CEU         0.788829  5.684941e-02  0.044110  0.110211\n",
       "CHB         0.104775  1.234433e-13  0.000228  0.894997\n",
       "CHS         0.117412  6.570803e-14  0.000253  0.882334\n",
       "CLM         0.590383  3.449625e-02  0.019500  0.355621\n",
       "ESN         0.071548  6.977994e-13  0.000085  0.928367\n",
       "FIN         0.812812  5.291742e-02  0.024798  0.109473\n",
       "GBR         0.789207  5.133869e-02  0.037341  0.122113\n",
       "GIH         0.198740  2.789187e-12  0.001089  0.800171\n",
       "GWD         0.077460  1.005816e-12  0.000120  0.922420\n",
       "IBS         0.723568  4.383728e-02  0.027501  0.205094\n",
       "ITU         0.165784  6.511905e-04  0.000727  0.832838\n",
       "JPT         0.151616  9.520952e-13  0.000230  0.848154\n",
       "KHV         0.120253  5.276638e-13  0.000334  0.879412\n",
       "LWK         0.080059  6.184784e-13  0.000110  0.919831\n",
       "MSL         0.101333  5.837016e-08  0.000120  0.898547\n",
       "MXL         0.402906  1.738450e-02  0.008188  0.571521\n",
       "PEL         0.230073  4.555846e-03  0.001854  0.763516\n",
       "PJL         0.160604  5.917921e-08  0.000739  0.838656\n",
       "PUR         0.490955  3.516436e-02  0.015530  0.458350\n",
       "STU         0.160275  1.486042e-03  0.001155  0.837084\n",
       "TSI         0.796755  6.499689e-02  0.039268  0.098980\n",
       "YRI         0.076407  9.138039e-13  0.000099  0.923494"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import math \n",
    "\n",
    "alpha1 = 1.70\n",
    "alpha2 = -1.26\n",
    "alpha3 = -2.62\n",
    "\n",
    "results = pd.DataFrame(columns = ['pBlond', 'pBrown', 'pRed', 'pBlack'], index = samples_list,dtype=float)\n",
    "\n",
    "for sample in samples_list: #for each sample\n",
    "    \n",
    "    table =  all_GT[all_GT['sample'] == sample]  #only select values for that sample \n",
    "    sum_b1 = 0\n",
    "    sum_b2 = 0\n",
    "    sum_b3 = 0\n",
    "    rsid_R = ['rs1805006','rs11547464','rs1805007','rs1805008','rs1805009']\n",
    "    rsid_r = ['rs1805005','rs2228479','rs1110400','rs885479']\n",
    "    X_R = 0\n",
    "    X_r = 0\n",
    "\n",
    "    #first calculate a cumulative X for the MC1R_R and MC1R_r\n",
    "    count_hom = 0 \n",
    "    count_heter = 0\n",
    "    count_hom_alt = 0\n",
    "    for rsid in rsid_R:\n",
    "        if table[table['rsid'] == rsid].X.iloc[0] == 0:\n",
    "            count_hom = count_hom + 1\n",
    "        elif table[table['rsid'] == rsid].X.iloc[0] == 1:\n",
    "            count_heter = count_heter+1\n",
    "        else: #if X = 2 \n",
    "            count_hom_alt = count_hom_alt +1\n",
    "            \n",
    "    if count_hom == len(rsid_r):\n",
    "        X_R = 0\n",
    "    elif count_hom_alt >= 1:\n",
    "        X_R = 2\n",
    "    elif count_heter == 1:\n",
    "        X_R = 1\n",
    "    else:\n",
    "        X_R = 2\n",
    "        \n",
    "    #first calculate a cumulative X for the MC1R_R and MC1R_r\n",
    "    count_hom = 0 \n",
    "    count_heter = 0\n",
    "    count_hom_alt = 0\n",
    "    for rsid in rsid_r:\n",
    "        if table[table['rsid'] == rsid].X.iloc[0] == 0:\n",
    "            count_hom = count_hom +1\n",
    "        elif table[table['rsid'] == rsid].X.iloc[0]  == 1:\n",
    "            count_heter = count_heter+1\n",
    "        else: #if X = 2 \n",
    "            count_hom_alt = count_hom_alt +1\n",
    "            \n",
    "    if count_hom == len(rsid_r):\n",
    "        X_r = 0\n",
    "    elif count_hom_alt >= 1:\n",
    "        X_r = 2\n",
    "    elif count_heter == 1:\n",
    "        X_r = 1\n",
    "    else:\n",
    "        X_r = 2\n",
    "\n",
    "    for i,rid in enumerate(bed_file_concentrated['rsid']):  #go through SNPs\n",
    "    # I would add MC1R_R & MC1R_r to the bed file \n",
    "    # in here I would probably add an if statement for MC1R_R & MC1R_r\n",
    "    # and then would iterate through the list of rsids for each of those \n",
    "    # and decide value of X based on that \n",
    "    # would probably make a series of if statements (basically if you find homozygous, X = 2, etc)\n",
    "        if rid != 'MC1R_R' and rid != 'MC1R_r':\n",
    "            rsid = bed_file_concentrated['rsid'].iloc[i]\n",
    "            beta1 = bed_file_concentrated['1'].iloc[i]\n",
    "            beta2 = bed_file_concentrated['2'].iloc[i]\n",
    "            beta3 = bed_file_concentrated['3'].iloc[i]\n",
    "            X = table[table['rsid'] == rid].X.iloc[0]\n",
    "            sum_b1 = sum_b1 + beta1 * X\n",
    "            sum_b2 = sum_b2 + beta2 * X\n",
    "            sum_b3 = sum_b3 + beta3 * X\n",
    "        elif i == 'MC1R_R':\n",
    "            rsid = bed_file_concentrated['rsid'].iloc[i]\n",
    "            beta1 = bed_file_concentrated['1'].iloc[i]\n",
    "            beta2 = bed_file_concentrated['2'].iloc[i]\n",
    "            beta3 = bed_file_concentrated['3'].iloc[i]\n",
    "            X = X_R #calculated before\n",
    "            sum_b1 = sum_b1 + beta1 * X\n",
    "            sum_b2 = sum_b2 + beta2 * X\n",
    "            sum_b3 = sum_b3 + beta3 * X\n",
    "        elif i == 'MC1R_r':\n",
    "            rsid = bed_file_concentrated['rsid'].iloc[i]\n",
    "            beta1 = bed_file_concentrated['1'].iloc[i]\n",
    "            beta2 = bed_file_concentrated['2'].iloc[i]\n",
    "            beta3 = bed_file_concentrated['3'].iloc[i]\n",
    "            X = X_r #calculated before\n",
    "            sum_b1 = sum_b1 + beta1 * X\n",
    "            sum_b2 = sum_b2 + beta2 * X\n",
    "            sum_b3 = sum_b3 + beta3 * X\n",
    "            \n",
    "    results.loc[sample, 'pBlond'] = (math.exp(alpha1+sum_b1))/(1+math.exp(alpha1+sum_b1)+math.exp(alpha2+sum_b2)+math.exp(alpha3+sum_b3))\n",
    "    results.loc[sample, 'pBrown'] = (math.exp(alpha2+sum_b2))/(1+math.exp(alpha1+sum_b1)+math.exp(alpha2+sum_b2)+math.exp(alpha3+sum_b3))\n",
    "    results.loc[sample, 'pRed'] = (math.exp(alpha3+sum_b3))/(1+math.exp(alpha1+sum_b1)+math.exp(alpha2+sum_b2)+math.exp(alpha3+sum_b3))\n",
    "    results.loc[sample, 'pBlack'] = 1 - results.loc[sample, 'pBlond'] - results.loc[sample, 'pBrown'] - results.loc[sample, 'pRed']\n",
    "    \n",
    "#the results dataframe contains all the hair color probabilities for all samples with \n",
    "#the corresponding ancestry information\n",
    "\n",
    "results.insert(0, 'population',sample_pop_match)\n",
    "results = results.dropna()\n",
    "results.groupby(['population']).mean()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "informative-campbell",
   "metadata": {},
   "outputs": [],
   "source": [
    "results.to_csv('results.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "affecting-lyric",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
